libs/agno/agno/compression/manager.py

```diff
@@ -49,32 +49,47 @@ class CompressionManager:
     model: Optional[Model] = None
     compress_tool_results: bool = True
     compress_tool_results_limit: int = 3
+    compress_tool_results_token_limit: Optional[int] = None  # Compress when token count >= this limit
     compress_tool_call_instructions: Optional[str] = None
 
     stats: Dict[str, Any] = field(default_factory=dict)
 
     def _is_tool_result_message(self, msg: Message) -> bool:
         return msg.role == "tool"
 
-    def should_compress(self, messages: List[Message]) -> bool:
+    def should_compress(self, messages: List[Message], tools: Optional[List] = None) -> bool:
         if not self.compress_tool_results:
             return False
 
+        # Token-based threshold check
+        if self.compress_tool_results_token_limit is not None and self.model is not None:
```

Comment:
Is self.model ever None? Probably the CompressionManager doesn't make sense without it?

Reply:
But this doesn't make sense right? Here you are counting tokens against the compression manager's model, which could be different from your main model... you need to count against your main model's?